### PR TITLE
#381 improve interpolation behaviour

### PR DESCRIPTION
**Pull request by kfriedberger**
see #381 for details

### CODE REVIEW

File: src/org/sosy_lab/java_smt/test/InterpolatingProverTest.java [hunk 1]
```diff
@@ -1179,4 +1179,70 @@ public <T> void testInvalidToken() throws InterruptedException, SolverException
     assertThrows(
         IllegalArgumentException.class, () -> stack.getInterpolant(ImmutableList.of((T) p3)));
   }
+
+  @Test
+  @SuppressWarnings("unused")
```
Comment (baierd) [inline src/org/sosy_lab/java_smt/test/InterpolatingProverTest.java:1184]: The unused suppression's should be removed ;D

File: src/org/sosy_lab/java_smt/solvers/cvc5/CVC5AbstractProver.java [hunk 2]
```diff
@@ -103,17 +113,20 @@ protected void popImpl() {
...
+  @CanIgnoreReturnValue
+  protected String addConstraint0(BooleanFormula pF) {
     Preconditions.checkState(!closed);
     setChanged();
     Term exp = creator.extractInfo(pF);
     if (incremental) {
       solver.assertFormula(exp);
     }
-    return null;
+    String id = "ID_" + ID_GENERATOR.getFreshId();
```
Comment (baierd) [inline src/org/sosy_lab/java_smt/solvers/cvc5/CVC5AbstractProver.java:127]: I totally get what you are doing here. But we should create a new class (InterpolationPoint?) that wraps this ID to make the intention clear for user and future developers.

**Review by baierd — CHANGES_REQUESTED** [review]
This fixes the issue nicely. And besides the 2 small change requests this can be merged with no issues.

File: src/org/sosy_lab/java_smt/solvers/cvc5/CVC5AbstractProver.java (hunk 2 above)
Comment (kfriedberger) [reply src/org/sosy_lab/java_smt/solvers/cvc5/CVC5AbstractProver.java:127]: InterpolationPoint is a quite special use-case. Some solvers also use those references for unsat-cores or model generation. We can think about it independently of this issue. Maybe name the reference to something like FormulaBacktrackPoint.

**Pull request merged**